File: isma-intg-api/src/main/java/ru/nstu/isma/intg/api/IntgResultMemoryStore.java
package ru.nstu.isma.intg.api;

import java.util.LinkedList;
import java.util.List;
import java.util.function.Consumer;

public class IntgResultMemoryStore implements Consumer<IntgResultPoint>, IntgResultPointProvider {

    private final List<IntgResultPoint> results = new LinkedList<>();

    @Override
    public void accept(IntgResultPoint intgResultPoint) {
        results.add(intgResultPoint);
    }

    @Override
    public void read(Consumer<List<IntgResultPoint>> handler) {
        handler.accept(results);
    }

}

File: isma-intg-api/src/main/kotlin/ru/nstu/isma/intg/api/IntgResultMemoryStore.kt
package ru.nstu.isma.intg.api

import kotlinx.coroutines.flow.asFlow
import java.util.*
import java.util.function.Consumer

class IntgResultMemoryStore : IntgResultPointProvider {
    private val resultsInternal = LinkedList<IntgResultPoint>()

    override val results get() = resultsInternal.asFlow()

    fun accept(intgResultPoint: IntgResultPoint) {
        resultsInternal.add(intgResultPoint)
    }

    fun read(handler: Consumer<List<IntgResultPoint>>) {
        handler.accept(resultsInternal)
    }
}